generator/templates/Deployment/_github/workflows/deploy.test.yml

```diff
@@ -53,11 +54,48 @@ jobs:
           <%_ if (options.plugins.includes('sentry')) { _%>
           echo "VUE_APP_ANALYTICS=" >> .env.production
           <%_ } _%>echo "VUE_APP_PUBLIC_PATH=/" >> .env.production
+      <%_ if (!options.addRobotsFile){ _%>
+      - name: setup robots.txt for disallowing search engines to visit the test env
+        run: |
+          cd public
+          touch robots.txt
+          echo "" > robots.txt
+          echo "User-agent: *" >> robots.txt
+          echo "Disallow: /" >> robots.txt
+      <%_ } _%>
       - name: build the environment
         run: |
           npm ci
           npm run build
+      - name: remove /old deployment and create /new folder for deployment
+          uses: appleboy/ssh-action@master
+          with:
+            host: host
+            username: user
+            key: ${{ secrets.private_key }}
+            proxy_host: jumphost.kingscode.nl
+            proxy_username: jumphost
+            proxy_key: ${{ secrets.private_key }}
+            script: |
+              cd ~/domains/example.com/public_html
+              rm -rf ./old
+              mkdir -p ./new
+              mkdir -p ./old
       - name: deployment
         run: |
           cd dist
-          rsync -ave "ssh -o ProxyJump=jumphost@jumphost.kingscode.nl" ./ user@host:~/domains/example.com/public_html
+          rsync -ave "ssh -o ProxyJump=jumphost@jumphost.kingscode.nl" ./ user@host:~/domains/example.com/public_html/new
+      - name: move previous deployment to ./old and move ./new deployment to ./
```

Comment:
Why is this split up in 2 actions?

You don't need to create a `new` directory for the sake of directly copying everything from it in the end and then removing the directory. The html root _is_ the new directory.

-> create old dir
-> move to old dir
-> clear target dirs
-> copy dist to html root dir

Reply:
While moving to old dir there will be no content available for a short period of time, we want no downtime deployments.

Reply:
And it prevents failures, when we remove all the old files first and the build fails, there is nothing lost.

Reply:
Not like that would happen, because the action won't run if it fails